src/test/java/io/vertx/redis/client/impl/RedisURITest.java

```diff
@@ -33,6 +36,14 @@ public void testTwoPasswordsAreGiven() {
     Assert.assertEquals("Password is not correct", "pass", redisURI.password());
   }
 
+  @Test
+  public void testPasswordWithEncodeInQuery() throws UnsupportedEncodingException {
```

Comment:
This test passes even without the fix. Actually I believe this is not a bug.

Reply:
have you run the test? a new test was commit.you can review it

Reply:
After contact offline, I understand the problem now. Yes it should be a bug in the old API :)